### DISCUSSION TITLE
Superset connectivity with clickhouse

### DISCUSSION THREAD

**Discussion by sanjaypocket52** [apache/superset#10700] [General]
How to connect clickhouse with superset clickhouse://default@IP:9000/data_lake

It is gave me below error
ERROR: Could not load database driver: clickhouse

while i have installed pip install clickhouse-sqlalchemy

**Comment by pesto93** [comment]
Did you figure it out? I am having same problem here.

**Comment by anilvpatel21** [comment]
I need to restart the superset. Do you know how to connect with multiple hosts?

**Comment by Teeeto** [comment]
I had no luck with clickhouse-jdbc driver
However clickhouse4j worked like a charm
https://github.com/blynkkk/clickhouse4j
The URL should be like clickhouse+native://default@host/database

**Reply by Mehdi-YC** [reply]
hello sir , did you connect clickhouse to superset with the java connector ? i need to get the  clickhouse+native dialect working

**Reply by Slach** [reply]
@SAVE-POlNT 
you don't need to clickhouse4j to connect to clickhouse
latest version of https://github.com/xzkostyan/clickhouse-sqlalchemy works fine

**Reply by Mehdi-YC** [reply]
well , it works but not that fine , you can not get data from your views , it returns no rows

**Reply by Slach** [reply]
@SAVE-POlNT could you share SQL query which show "no rows" for you?
Do you use some complex data types like enum, multi dimensional arrays or nested?

**Reply by Mehdi-YC** [reply]
no i just partitionnedemy table and quueried with select * from table_name in chart interface

**Reply by Slach** [reply]
could you share `DESCRIBE your_db.table_name;` results?

**Reply by Mehdi-YC** [reply]
im sorry , my ssh died yesterday , so i lost my data , basically i had 2 columns , it was a test database one of them was Decimal10(3)
and im pretty sure the problem was from it  the other oner was only a String and worked for other tables 

the 2nd problem is when i partition any table , i get only the first partition back when i query or want to add them to raw records chart

**Comment by l10178** [comment]
I get the same error, but when I refresh the pages, the database has been added and it works!

**Reply by Mehdi-YC** [reply]
do you get all the rows in graph explorer ?
i partitioned my table by date , and im getting only the first partition back as result

**Reply by l10178** [reply]
Yes, I got all.
